src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -115,6 +115,24 @@ Container<T> add(const Container<T>& a, const Container<T>& b, T alpha=1, T beta
 }
 
 /**
+ * Performs the operation C = alpha*A + beta*B.
+ * @param A first matrix
+ * @param B second matrix
+ * @param alpha constant to be multiplied by the first matrix
+ * @param beta constant to be multiplied by the second matrix
+ * @return The result matrix
+ */
+template <typename T>
+SGMatrix<T> add(const SGMatrix<T>& a, const SGMatrix<T>& b, T alpha=1, T beta=1)
+{
+	REQUIRE((a.num_rows == b.num_rows), "Matrices should have the same row number! "
```

Comment:
Can we do this less verbosy?
"Number of rows of matrix a (%d) must match matrix b (%d)"